code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -245,3 +329,232 @@
+	if(!isliving(target))
+		return ..()
+	var/mob/living/living_victim = target
+	if(living_victim.IsParalyzed())
+		say("GET OUT OF MY WAY, MEATBAG.")
+		enemies = list()
+		podspawn(list(
+			"target" = get_turf(living_victim),
+			"style" = STYLE_MISSILE,
+			"effectMissile" = TRUE,
+			"explosionSize" = list(0,0,1,2),
+			"delays" = list(POD_TRANSIT = 3 SECONDS, POD_FALLING = 0.5 SECONDS, POD_OPENING = 0, POD_LEAVING = 0)
+		))
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/retaliate/trader/amorphous/AttackingTarget(atom/attacked_target)
+	OpenFire(target)
+
+///base prototype of the pirates that handles lines they all have.
+/mob/living/simple_animal/hostile/retaliate/trader/pirate
+	name = "Pirate"
+	desc = "A space pirate looking to flip some mysteriously procured wares for a quick buck."
+	icon_state = "piratemerchant"
+	loot = list(/obj/effect/mob_spawn/human/corpse/pirate)
+	faction = list("pirate")
+	itemsellcancelphrase = "Fine, keep it. But you should know i'm the only one here who will take it."
+	buyphrase = "Heh, here you go. No refunds!"
+	nocashphrase = "Hey, this isn't a giveaway! Earn some credits or don't come back."
+	///Phrase said when entering an off limits area
+	var/offlimitsphrase = "Hey, get out of here! Off limits!"
+	///Phrase said when leaving an off limits area
+	var/exitofflimitsphrase = "That's right, stay on YOUR side of the counter."
+	lore = list(
+		"A pirate's life for me... er, no, I'm not a pirate. Who's asking?",
+		"We need some damn cryopods on this ship. You know how boring it gets when the FTL is long distance?",
+		"Nanotrasen's funny. In the core sectors, they'll blow your ship apart because it isn't painted white. \
+		But out here? Way different vibe.",
+		"This old junker is originally a transport ship, until we acquired it. Legally, though.",
+		"If we get enough credits, maybe we can get one of those ship weapons working.",
+		"Have you heard of the Silver-Scales? They go around demanding tribute from stations like yours."
+	)
+
+/mob/living/simple_animal/hostile/retaliate/trader/pirate/offlimits_enter_reaction(datum/source, mob/living/arriving)
+	. = ..()
+	if(!.)
+		return
+	if(!faction_check_mob(arriving))
+		INVOKE_ASYNC(src, .proc/aggro_intruder, arriving)
+
+/mob/living/simple_animal/hostile/retaliate/trader/pirate/offlimits_exit_reaction(datum/source, mob/living/leaving)
+	. = ..()
+	if(!.)
+		return
+	if(!faction_check_mob(leaving))
+		INVOKE_ASYNC(src, .proc/deaggro_intruder, leaving)
+
+///small helper proc that does the part of the area enter signal that sleeps
+/mob/living/simple_animal/hostile/retaliate/trader/pirate/proc/aggro_intruder(mob/living/intruder)
+	say(offlimitsphrase)
+	enemies |= WEAKREF(intruder)
+
+///small helper proc that does the part of the area exit signal that sleeps
+/mob/living/simple_animal/hostile/retaliate/trader/pirate/proc/deaggro_intruder(mob/living/intruder)
+	say(exitofflimitsphrase)
+	enemies &= ~WEAKREF(intruder)
```

Comment:
```suggestion
	enemies -= WEAKREF(intruder)
```

Reply:
I'm not even sure your original code works considering WEAKREF isn't a bitfield.